package com.eolwral.osmonitor.core;

message processInfo {
  required string name = 1;           /**< process name */
  required string owner = 2;          /**< process owner */

  enum processStatus {
    Unknown = 0;    /**< process status is unknown */
    Running = 1;    /**< process is running */
    Sleep = 2;      /**< process is s sleeping in an interruptible wait */
    Stopped = 3;    /**< process is traced or stopped */
    Disk = 4;       /**< process is is waiting in uninterruptible disk sleep */
    Zombie = 5;     /**< process is zombie that couldn't be killed */
    Page = 6;       /**< process is paging */ 
  } 
  required processStatus Status = 3 [default = Unknown];  /**< process status */

  required uint32 uid = 4;            /**< process uid */
  required uint32 pid = 5;            /**< process pid */
  required uint32 ppid = 6;           /**< process pid for parent */
  required uint64 rss = 7;            /**< resident set size */
  required uint64 vsz = 8;            /**< virtual size */
  required float  cpuUsage = 9;       /**< CPU usage */
  required uint32 threadCount = 10;   /**< thread count for this process */
  required uint32 priorityLevel = 11; /**< priority from -20 to 20 */
  required uint64 usedUserTime = 12;  /**< used user time (user mode) */
  required uint64 usedSystemTime = 13; /**< used system time (kernel mode) */
  required uint64 startTime = 14;     /**< start time */
  required uint64 cpuTime = 15;       /**< CPU time */
}